File: app/src/main/java/com/example/communityeventmanagement/domain/entities/Community.kt
package com.example.communityeventmanagement.domain.entities

data class Community(
    val id: Int,
    val name: String,
    val description: String,
    val category: String,
    val coverImageUri: String? = null,
    val organizerId: String,
    val organizerName: String,
    val memberCount: Int = 0,
    val memberIds: List<String> = emptyList(),
    val events: List<Event> = emptyList(),
    val forumMessages: List<ForumMessage> = emptyList(),
) {
}

data class ForumMessage(
    val sender: String,
    val message: String,
    val time: String,
    val avatarInitials: String,
)

File: communityApp-Backend/app/Models/Community.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Community extends Model
{
    protected $fillable = [
        'name',
        'description',
        'organizer_id',
        'category_id',
        'status',
        'cover_image_url',
        'member_count'
    ];

    public function organizer()
    {
        return $this->belongsTo(
            User::class,
            'organizer_id'
        );
    }

    public function category()
    {
        return $this->belongsTo(
            Category::class
        );
    }

    public function members()
    {
        return $this->belongsToMany(
            User::class,
            'community_members'
        )->withPivot('role', 'joined_at');
    }

    public function events()
    {
        return $this->hasMany(
            Event::class
        );
    }

    public function forumMessages()
    {
        return $this->hasMany(
            ForumMessage::class
        );
    }
}